pkg/util/util_test.go

```diff
@@ -1127,3 +1129,31 @@ func TestRemoveRelativePathFromFiles(t *testing.T) {
 		})
 	}
 }
+
+func TestHttpGetFreePort(t *testing.T) {
+	tests := []struct {
+		name    string
+		wantErr bool
+	}{
+		{
```

Comment:
I don't think it would be a simple OS independent solution to block all the ports or generate a scenario to throw an error while closing the listener. If you have some ideas regarding this, please share them.